scipy/spatial/tests/test_slerp.py

```diff
@@ -0,0 +1,378 @@
+                                          end=end,
+                                          t=forward_t_vals)
+        reverse_results = geometric_slerp(start=start,
+                                          end=end,
+                                          t=reverse_t_vals)
+        scrambled_results = geometric_slerp(start=start,
+                                            end=end,
+                                            t=scramble_t_vals)
+
+        # check fidelity to input order
+        assert_allclose(forward_results, np.flipud(reverse_results))
+        assert_allclose(forward_results[shuffled_indices],
+                        scrambled_results)
+
+    @pytest.mark.parametrize("t", [
+        # string:
+        "15, 5, 7",
+        # complex numbers currently produce a warning
+        # but not sure we need to worry about it too much:
+        # [3 + 1j, 5 + 2j],
+        ])
+    def test_t_values_conversion(self, t):
+        with pytest.raises(ValueError):
+            scrambled_results = geometric_slerp(start=np.array([1]),
+                                                end=np.array([0]),
+                                                t=t)
+
+    def test_accept_arraylike(self):
+        # array-like support requested by reviewer
+        # in gh-10380
+        actual = geometric_slerp([1, 0], [0, 1], [0, 1/3, 0.5, 2/3, 1])
+
+        # expected values are based on visual inspection
+        # of the unit circle for the progressions along
+        # the circumference provided in t
+        expected = np.array([[1, 0],
+                             [np.sqrt(3) / 2, 0.5],
+                             [np.sqrt(2) / 2,
+                              np.sqrt(2) / 2],
+                             [0.5, np.sqrt(3) / 2],
+                             [0, 1]], dtype=np.float64)
+        # Tyler's original Cython implementation of geometric_slerp
+        # can pass at atol=0 here, but on balance we will accept
+        # 1e-16 for an implementation that avoids Cython and
+        # makes up accuracy ground elsewhere
+        assert_allclose(actual, expected, atol=1e-16)
+
+    def test_scalar_t(self):
+        # when t is a scalar, return value is a single
+        # interpolated point of the appropriate dimensionality
+        # requested by reviewer in gh-10380
+        actual = geometric_slerp([1, 0], [0, 1], 0.5)
+        expected = np.array([np.sqrt(2) / 2,
+                             np.sqrt(2) / 2], dtype=np.float64)
+        assert actual.shape == (2,)
+        assert_allclose(actual, expected)
+
+    def test_degenerate_input(self):
+        # handle start == end with ValueError
+        start = np.array([1, 0, 0])
+        with pytest.raises(ValueError, match="cannot be the same"):
+            geometric_slerp(start=start,
+                            end=start,
+                            t=np.linspace(0, 1, 5))
+
+    @pytest.mark.parametrize('k',
+    np.logspace(-10, -1, 10),
```

Comment:
`pycodestyle` doesn't like this line, but I haven't tried to figure out what the "correct" spacing should be to avoid over-indent or under-indent issues from `pycodestyle`.  In this case, the entire decorator easily fits in one line, so why not just write

    @pytest.mark.parametrize('k', np.logspace(-10, -1, 10))